index.js

```diff
@@ -141,44 +178,63 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+let ticketNumber = 0
+function takeATicketNumber(line){
+  ticketNumber++
+  line.push(ticketNumber)
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+
+
+const newLine = []
+
+function ticketNumberGeneratorFunc(){
+  let ticketNumber = 0
+
+  return function something() {
+    ticketNumber++
+    return `Welcome. You are ticket number ${ticketNumber}`
+  }
+
+}
```

Comment:
Close! I don't think you need to define the function as 'something' but rather as an anonymous function.

```
function ticketNumberGeneratorFunc() {
  let number = 0;
  return (function (line) {
    number += 1;
    line.push(number);
    return `Welcome. You are ticket number ${number}`;
  })
}
```

Don't forget to add the ticket number to the line. You need to be able to pass in the line as an argument to the anonymous function. That way this function can generate tickets for any given line.